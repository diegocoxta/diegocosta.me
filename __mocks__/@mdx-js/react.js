module.exports = {
  MDXProvider: jest.fn(({ children }) => children),
};
