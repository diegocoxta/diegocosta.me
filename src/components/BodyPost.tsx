import styled from 'styled-components';

export interface BodyPostProps {
  html: string;
}

export default styled.section.attrs<BodyPostProps>((props) => ({
  dangerouslySetInnerHTML: { __html: props.html },
}))<BodyPostProps>`
  line-height: 1.5;
  font-size: 19px;
  margin-top: 10px;

  a {
    color: ${({ theme }) => theme.accentColor};
    text-decoration: none;
    border-bottom: 1px solid ${({ theme }) => theme.backgroundColor};

    &:hover,
    &:focus {
      border-bottom: 1px solid ${({ theme }) => theme.accentColor};
      outline: none;
    }
  }

  code {
    background: ${({ theme }) => theme.accentColor}40;
    padding: 2px 6px;
    border-radius: 10px;
    font-size: 16px;
  }

  blockquote {
    border-left: 5px solid ${({ theme }) => theme.accentColor};
    padding-left: 20px;
    margin-left: 0;
  }

  img {
    box-shadow: none !important;
  }

  h3 {
    font-size: 28px;
  }

  table.uses-table {
    width: 100%;
    margin: 40px 0;

    td {
      text-align: center;
    }
  }
`;
