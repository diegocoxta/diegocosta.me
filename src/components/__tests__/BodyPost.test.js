import React from 'react';
import { render } from '@testing-library/react';

import BodyPost from '../BodyPost';

describe('<BodyPost />', () => {
  it('renders properly', () => {
    const { baseElement, getByText } = render(<BodyPost html="My page BodyPost" />);

    expect(getByText('My page BodyPost')).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });
});
