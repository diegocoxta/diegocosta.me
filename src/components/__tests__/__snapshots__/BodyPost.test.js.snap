// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BodyPost /> renders properly 1`] = `
.c0 {
  line-height: 1.5;
  font-size: 19px;
  margin-top: 10px;
}

.c0 a {
  text-decoration: none;
  border-bottom: 1px solid;
}

.c0 a:hover,
.c0 a:focus {
  border-bottom: 1px solid;
  outline: none;
}

.c0 code {
  background: 40;
  padding: 2px 6px;
  border-radius: 10px;
  font-size: 16px;
}

.c0 blockquote {
  border-left: 5px solid;
  padding-left: 20px;
  margin-left: 0;
}

.c0 img {
  box-shadow: none!important;
}

.c0 h3 {
  font-size: 28px;
}

.c0 table.uses-table {
  width: 100%;
  margin: 40px 0;
}

.c0 table.uses-table td {
  text-align: center;
}

<body>
  <div>
    <section
      class="c0"
      html="My page BodyPost"
    >
      My page BodyPost
    </section>
  </div>
</body>
`;
